ERCS/erc-7527.md

```diff
@@ -0,0 +1,516 @@
+ */
+struct AppSettings {
+    address implementation;
+    bytes immutableData;
+    bytes initData;
+}
+
+interface IERC7527Factory {
+    /**
+     * @dev Deploys a new agency and app clone and initializes both.
+     * @param agencySettings The settings of the agency.
+     * @param appSettings The settings of the app.
+     * @param data The data is additional data, it has no specified format and it is sent in call to `factory`.
+     * @return appInstance The address of the created proxy contract of appImplementation.
+     * @return agencyInstance The address of the created proxy contract of agencyImplementation.
+     */
+    function deployWrap(AgencySettings calldata agencySettings, AppSettings calldata appSettings, bytes calldata data)
+        external
+        returns (address, address);
+}
+
+```
+
+## Rationale
+
+### Prior Interfaces
+
+[ERC-5679](./eip-5679.md) proposed `IERC5679Ext721` interface for introducing a consistent way to extend [ERC-721](./eip-721.md) token standards for minting and burning. To ensure the backward compatibility, considering some contracts which do not implement `ERC721TokenReceiver`, `IERC7527App` employ `mint` function instead of `safeMint`. To ensure the safety and the uniqueness of mutual bound, the `_from` parameter of the `burn` function in `IERC5679Ext721` must be the contract address of the bounded angency. Thus, `burn` function in `IERC7527App` does not contain the `_from` parameter. 
+
+### Mutual Bound
+
+Implement contracts for `IERC7527App` and `IERC7527Agency` so that they are each other's only owner. The wrap process is to check the premium amount of the fungible token received and then mint non-fungible token in the App. Only the owner or an approver of the non-fungible token can unwrap it.
+
+### Implementation Diversity 
+
+Users can customize function and fee percentage when implement the Agency and the App interfaces.
+
+Different Agency implementations have distinct wrap, unwrap function logic, and different oracleFunction. Users can customize the currency, initial price, fee receiving address, fee rate, etc., to initialize the Agency contract. 
+
+Different App implementations cater to various use cases. Users can initialize the App contract.
+
+Factory is not required. Factory implementation is need-based. Users can deploy their own contracts by selecting different Agency implementations and different App implementations through the Factory, combining them to create various products.
+
+
+### Currency types
+
+Support ETH and [ERC-20](./eip-20.md).
+
+### [ERC-721](./eip-721.md) - Required for ID
+
+For each wrap process, a unique ID should be generated. This ID is essential for verification during the unwrap process. It also serves as the exclusive credential for the token. This mechanism ensures the security of assets within the Agency. Although the ID is not mandatory to use the [ERC-721](./eip-721.md) standard, `name` MUST follow the standard from `ERC721Metadata`. 
```

Comment:
> Although the ID is not mandatory to use the [ERC-721](./eip-721.md) standard

This is unclear. Are you saying that the ID does not need to be a token id from ERC-721, or something else?

Reply:
We only want to emphasize uniqueness of token id. Maybe mentioning ERC-721 is a bit misleading. We've deleted this sentence.